src/mtype.d

```diff
@@ -7819,7 +7816,8 @@ public:
              * (e.field0, e.field1, e.field2, ...)
              */
             e = e.semantic(sc); // do this before turning on noaccesscheck
-            e.type.size(); // do semantic of type
+
+            sym.size(e.loc); // do semantic of type
```

Comment:
The necessary thing to do here is, to call `StructDeclaration.size`. Even if `e.type` is Tstruct (e.g. `e.type.ty == Tenum && e.type.tobasetype().ty == Tstruct`?), we don't have to call `e.type.size` in here.